mangaki/mangaki/algo/als3.py

```diff
@@ -0,0 +1,82 @@
+from mangaki.algo.recommendation_algorithm import RecommendationAlgorithm, register_algorithm
+from scipy.sparse import coo_matrix
+from collections import defaultdict
+import numpy as np
+
+
+@register_algorithm('als3', {'nb_components': 20})
+class MangakiALS3(RecommendationAlgorithm):
+    def __init__(self, nb_components=20, nb_iterations=20, lambda_=0.1):
+        super().__init__()
+        self.nb_components = nb_components
+        self.nb_iterations = nb_iterations
+        self.lambda_ = lambda_
+
+    @property
+    def is_serializable(self):
+        return True
+
+    def fit(self, X, y):
+        # self.X_test = X_test
+        # self.y_test = y_test
+        self.init_vars()
+        self.bias = y.mean()
+        self.matrix, self.matrixT = self.to_dict(X, y)
+        self.ratings_of_user, self.ratings_of_work = self.to_sparse(X, y)
+        users, works = map(np.unique, self.ratings_of_user.nonzero())
+        for nb_iter in range(self.nb_iterations):
+            # print('Step', nb_iter, self.compute_rmse(self.y_test, self.predict(self.X_test)))
+            for user_id in users:
+                self.fit_user(user_id)
+            for work_id in works:
+                self.fit_work(work_id)
+
+    def to_dict(self, X, y):
```

Comment:
Sounds like a utility for all of our ALS version, might be good to factor it out of ALS or in an "ALS-shared utilities" file.

Reply:
I see. I was sad because I thought it could be replaced by scipy sparse matrices, but these ones are way slower :'(